src/jit/lsraxarch.cpp

```diff
@@ -2297,224 +2297,246 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
+                srcCount += BuildOperandUses(op1);
+                srcCount += BuildOperandUses(op2);
+                srcCount += BuildOperandUses(op3, RBM_EDI);
+
                 buildUses = false;
-                srcCount  = 3;
+                break;
             }
-            break;
 
-        case NI_SSE41_TestAllOnes:
-        {
-            buildInternalFloatRegisterDefForNode(intrinsicTree);
-            break;
-        }
-
-        case NI_SSE41_Extract:
-            if (baseType == TYP_FLOAT)
+            case NI_SSE41_BlendVariable:
             {
-                buildInternalIntRegisterDefForNode(intrinsicTree);
+                assert(numArgs == 3);
+                assert(isRMW);
+
+                if (!compiler->canUseVexEncoding())
+                {
+                    // SSE4.1 blendv* hardcode the mask vector (op3) in XMM0
+                    srcCount += BuildOperandUses(op1);
+                    srcCount += BuildDelayFreeUses(op2);
+                    srcCount += BuildDelayFreeUses(op3, RBM_XMM0);
+
+                    buildUses = false;
+                }
+                break;
             }
-#ifdef _TARGET_X86_
-            else if (varTypeIsByte(baseType))
+
+            case NI_SSE41_TestAllOnes:
             {
-                dstCandidates = allByteRegs();
+                buildInternalFloatRegisterDefForNode(intrinsicTree);
+                break;
             }
+
+            case NI_SSE41_Extract:
+            {
+                if (baseType == TYP_FLOAT)
+                {
+                    buildInternalIntRegisterDefForNode(intrinsicTree);
+                }
+#ifdef _TARGET_X86_
+                else if (varTypeIsByte(baseType))
+                {
+                    dstCandidates = allByteRegs();
+                }
 #endif
-            break;
+                break;
+            }
 
 #ifdef _TARGET_X86_
-        case NI_SSE42_Crc32:
-        {
-            // CRC32 may operate over "byte" but on x86 only RBM_BYTE_REGS can be used as byte registers.
-            //
-            // TODO - currently we use the BaseType to bring the type of the second argument
-            // to the code generator. May encode the overload info in other way.
-            var_types srcType       = intrinsicTree->gtSIMDBaseType;
-            regMaskTP op2Candidates = RBM_NONE;
-            BuildUse(op1);
-            if (varTypeIsByte(srcType))
-            {
-                op2Candidates = allByteRegs();
-            }
-            RefPosition* op2Use = BuildUse(op2, op2Candidates);
-            assert(isRMW);
-            setDelayFree(op2Use);
-            srcCount  = 2;
-            buildUses = false;
-            break;
-        }
-#endif // _TARGET_X86_
+            case NI_SSE42_Crc32:
+            {
+                // TODO - currently we use the BaseType to bring the type of the second argument
+                // to the code generator. May encode the overload info in other way.
```

Comment:
Nit: `TODO` comments should be qualified - here it should probably be `TODO-XArch-Cleanup` (see https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#7.1.5). Also, you could expand the last sentence to "We may want to encode the overload info in another way." to be clearer.